modules/Administration/GoogleCalendarAuthHandler.php

```diff
@@ -0,0 +1,180 @@
+*
+* You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+* SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+*
+* The interactive user interfaces in modified source and object code versions
+* of this program must display Appropriate Legal Notices, as required under
+* Section 5 of the GNU Affero General Public License version 3.
+*
+* In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+* these Appropriate Legal Notices must retain the display of the "Powered by
+* SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+* reasonably feasible for technical reasons, the Appropriate Legal Notices must
+* display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+*/
+
+if (!defined('sugarEntry') || !sugarEntry) {
+   die('Not A Valid Entry Point');
+}
+
+class GoogleCalendarAuthHandler
+{
+
+    /**
+     *
+     * @var string
+     */
+    protected $tplPath = '';
+
+    /**
+     *
+     * @var User
+     */
+    protected $currentUser = null;
+
+    /**
+     *
+     * @var array
+     */
+    protected $request = null;
+
+    /**
+     *
+     * @var array
+     */
+    protected $modStrings = null;
+
+    /**
+     *
+     * @var Configurator
+     */
+    protected $configurator = null;
+
+    /**
+     *
+     * @var Sugar_Smarty
+     */
+    protected $ss = null;
+
+    /**
+     *
+     * @var javascript
+     */
+    protected $js = null;
+
+
+
+    /**
+     * Setup object
+     *
+     * @param  User         $current_user
+     * @param               $request
+     * @param               $mod_strings
+     * @param  Configurator $config
+     * @param  Sugar_Smarty $sugar_smarty
+     * @param  javascript   $js
+     * @return void
+     */
+    public function __construct(string $tpl_path, User $current_user, $request, $mod_strings, Configurator $config, Sugar_Smarty $sugar_smarty, javascript $js)
+    {
+        $this->currentUser  = $current_user;
+        $this->request      = $request;
+        $this->modStrings   = $mod_strings;
+        $this->ss           = $sugar_smarty;
+        $this->tplPath      = $tpl_path;
+        $this->js           = $js;
+        $this->configurator = $config;
+
+        if (isset($this->request['do']) && $this->request['do'] == 'save') {
+            $this->configurator->config['google_auth_json'] = !empty($this->request['google_auth_json']);
+            $this->configurator->saveConfig();
+            header('Location: index.php?module=Administration&action=index');
+            exit();
+        }
+
+        $this->handleDisplay();
+    }
+
+
+    /**
+     * This function handles displaying the template
+     *
+     * @return void
+     */
+    protected function handleDisplay()
+    {
+        $errors = array();
+
+        // Check current user is admin
+        if (!is_admin($this->currentUser)) {
+            sugar_die("Unauthorized access to administration.");
```

Comment:
can you please separate out to a non-private method the die(), sugar_die() and exit calls.